components/style/values/specified/position.rs

```diff
@@ -366,3 +367,125 @@ impl ToCss for LegacyPosition {
         self.vertical.to_css(dest)
     }
 }
+
+#[derive(Clone, Copy, Debug, Eq, MallocSizeOf, PartialEq, ToComputedValue)]
+/// Auto-placement algorithm Option
+pub enum AutoFlow {
+    /// The auto-placement algorithm places items by filling each row in turn,
+    /// adding new rows as necessary.
+    Row,
+    /// The auto-placement algorithm places items by filling each column in turn,
+    /// adding new columns as necessary.
+    Column,
+}
+
+#[derive(Clone, Copy, Debug, Eq, MallocSizeOf, PartialEq, ToComputedValue)]
+/// Controls how the auto-placement algorithm works
+/// specifying exactly how auto-placed items get flowed into the grid
+pub struct GridAutoFlow {
+    /// Specifiy how auto-placement algorithm fills each `row` or `column` in turn
+    pub autoflow: AutoFlow,
+    /// Specify use `dense` packing algorithm or not
+    pub dense: bool,
+}
+
+impl GridAutoFlow {
+    #[inline]
+    /// Get default `grid-auto-flow` as `row`
+    pub fn row() -> GridAutoFlow {
+        GridAutoFlow {
+            autoflow: AutoFlow::Row,
+            dense: false,
+        }
+    }
+}
+
+impl ToCss for GridAutoFlow {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        dest.write_str(match self.autoflow {
```

Comment:
Probably just `derive(ToCss)` for `AutoFlow` and do:

```rust
self.autoflow.to_css(dest)?;
if self.dense { .. }
```

Reply:
Let me try it, thanks!